### DISCUSSION TITLE
How to disable ASP HttpRequestIn Activity listeners?

### DISCUSSION THREAD

**Discussion by ogxd** [dotnet/aspnetcore#54551] [Q&A] [answered]
Hello,

It seems that since recent versions of dotnet, Activities named `Microsoft.AspNetCore.Hosting.HttpRequestIn` are created when using ASP NET Core. This is a problem for us because we've implemented tracing ourselves a while ago and we don't want automatic instrumentation (it does not fit all of our usecases).  

Our `ActivityListener`s responsible of exporting the Activities (as spans in the context of an opentelemetry collector based pipeline) filter on the source names already to only listen to sources we're interested in, yet however it seems some Activities are generated despite not explicitely listening to them, such as the `Microsoft.AspNetCore.Hosting.HttpRequestIn` ones. We would have expected `Activity.Current` to be null in an ASP NET core controller if there is no listener.

I am guessing there is some kind of default listener in ASP NET Core. Is it possible to disable them?

Thanks in advance!

**Comment by ogxd** [comment]
Hmmm it seems I was wrong, there are no listener by default:  
<img width="532" alt="image" src="https://github.com/dotnet/aspnetcore/assets/12692438/51e3cbb7-0e41-45bd-b405-d4da9ab51ad2">

But then I wonder why activity is not null

**Reply by ogxd** [reply, marked as answer]
Ok I found out, I had to disable a logging source (weird to me that has an impact on tracing, but anyway..). 
Here is the place in code where this activity is created despite having no listeners on that source: 
https://github.com/dotnet/aspnetcore/blob/6b7a6a94fb3ff2c004f354dbfe662644dac0ce26/src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs#L416-L418

In `appsettings.json`:
```json
"Logging": {
    "LogLevel": {
        "Microsoft.AspNetCore.Hosting.Diagnostics": "None"
    }
}
```

This was answered in this issue: https://github.com/dotnet/aspnetcore/issues/18955.
I didn't need the `EventLog` section in my case.

**Comment by gmkado** [comment]
For anyone using NLog, I had to add:

```json
"Logging": {
    "LogLevel": {
        "Microsoft.AspNetCore.Hosting.Diagnostics": "None"
    }
    "NLog.Extensions.Logging.NLogLoggerProvider": {
      "LogLevel": {
        "Microsoft.AspNetCore.Hosting.Diagnostics": "None"
      }
    }
}
```

Specifically, I had to include the provider name to override the log levels from my nlog.config

**Comment by ivangelemerov** [comment]
Hey all, quick question on this issue here. I had this recently, I couldn't figure out what to do to remove it. I tried the approach in the answer, even adding it to the builder with builder.Logging.AddFilter("Microsoft.AspNetCore.Hosting.Diagnostics", LogLevel.None);, neither worked for me.

I ended up using a sampler like so:
```csharp
public override SamplingResult ShouldSample(in SamplingParameters p)
{
    // Drop *only* the well-known framework inbound request Activities
    if (p.Kind == ActivityKind.Server &&
        (p.Name == "Microsoft.AspNetCore.Hosting.HttpRequestIn"))
    {
        return new SamplingResult(SamplingDecision.Drop);
    }

    return _inner.ShouldSample(p);
}
```
Anyone got any ideas why it didn't work out? Did it get changed recently?
